examples/Consumer/Program.cs

```diff
@@ -71,26 +67,29 @@ public static void Run_Poll(string brokerList, List<string> topics)
                 consumer.OnConsumeError += (_, msg)
                     => Console.WriteLine($"Error consuming from topic/partition/offset {msg.Topic}/{msg.Partition}/{msg.Offset}: {msg.Error}");
 
-                consumer.OnOffsetsCommitted += (_, commit) =>
-                {
-                    Console.WriteLine($"[{string.Join(", ", commit.Offsets)}]");
-
-                    if (commit.Error)
-                    {
-                        Console.WriteLine($"Failed to commit offsets: {commit.Error}");
-                    }
-                    Console.WriteLine($"Successfully committed offsets: [{string.Join(", ", commit.Offsets)}]");
-                };
+                consumer.OnOffsetsCommitted += (_, commit) 
+                    => Console.WriteLine(
+                            commit.Error
+                                ? $"Failed to commit offsets: {commit.Error}"
+                                : $"Successfully committed offsets: [{string.Join(", ", commit.Offsets)}]");
 
+                // Raised when the consumer is assigned a new set of partitions.
                 consumer.OnPartitionsAssigned += (_, partitions) =>
                 {
                     Console.WriteLine($"Assigned partitions: [{string.Join(", ", partitions)}], member id: {consumer.MemberId}");
+                    // If you don't add a handler to the OnPartitionsAssigned event,
+                    // the below .Assign call happens automatically. If you do, you
+                    // must call .Assign explicitly in order for the consumer to 
+                    // start consuming messages.
                     consumer.Assign(partitions);
                 };
 
+                // Raised when the consumer's current assignment set has been revoked.
                 consumer.OnPartitionsRevoked += (_, partitions) =>
                 {
                     Console.WriteLine($"Revoked partitions: [{string.Join(", ", partitions)}]");
+                    // If you don't add a handler to the OnPartitionsRevoked event,
+                    // the below .Unassign call happens automatically.
```

Comment:
same thing here though, Unassign must be called. Be explicit since it is important